src/backersManager/newAllocation.ts

```diff
@@ -0,0 +1,134 @@
+  backer.backerTotalAllocation = backerTotalAllocation;
+
+  backer.save();
+
+  const gaugeId = backerAddress.concat(event.params.gauge_);
+  let gauge = GaugeStakingHistory.load(gaugeId);
+  if (gauge == null) {
+    gauge = new GaugeStakingHistory(gaugeId);
+    gauge.gauge = event.params.gauge_;
+    gauge.accumulatedAllocationsTime = DEFAULT_BIGINT;
+    gauge.backer = backerAddress;
+  } else {
+    const lastStakedSeconds = event.block.timestamp.minus(
+      gauge.lastBlockTimestamp
+    );
+    gauge.accumulatedAllocationsTime = gauge.accumulatedAllocationsTime.plus(
+      gauge.allocation.times(lastStakedSeconds)
+    );
+  }
+  gauge.allocation = event.params.allocation_;
+  gauge.lastBlockNumber = event.block.number;
+  gauge.lastBlockTimestamp = event.block.timestamp;
+
+  gauge.save();
+}
+
+function _getPreviousAllocation(gaugeToBuilder: GaugeToBuilder, backer: Address): BigInt {
+  let previousAllocation = DEFAULT_BIGINT;
+  const backerToBuilder = BackerToBuilder.load(
+    backer.concat(gaugeToBuilder.builder)
+  );
+  if (backerToBuilder != null) {
+    previousAllocation = backerToBuilder.totalAllocation;
+  }
+
+  return previousAllocation;
+}
+
+function _handleBuilder(event: NewAllocationEvent): void {
+  const gaugeToBuilder = GaugeToBuilder.load(event.params.gauge_);
+  if (gaugeToBuilder == null) return;
+
+  const builder = Builder.load(gaugeToBuilder.builder);
+  if (builder == null) return;
+
+  const previousAllocation = _getPreviousAllocation(gaugeToBuilder, event.params.backer_);
+  builder.totalAllocation = builder.totalAllocation.minus(previousAllocation).plus(event.params.allocation_);
+
+  const gaugeContract = GaugeRootstockCollectiveContract.bind(event.params.gauge_);
+  builder.rewardShares = gaugeContract.rewardShares();
+
+  builder.save();
+}
+
+function _handleBacker(event: NewAllocationEvent): void {
+  let backer = Backer.load(event.params.backer_);
+  if (backer == null) {
+    backer = new Backer(event.params.backer_);
+    backer.totalAllocation = DEFAULT_BIGINT;
+    backer.isBlacklisted = false;
+  }
+  const gaugeToBuilder = GaugeToBuilder.load(event.params.gauge_);
+  if (gaugeToBuilder == null) return;
+
+  const previousAllocation = _getPreviousAllocation(gaugeToBuilder, event.params.backer_);
+  backer.totalAllocation = backer.totalAllocation.minus(previousAllocation).plus(event.params.allocation_);
+
+  backer.save();
+}
+
+function _handleBackerToBuilder(event: NewAllocationEvent): void {
+  const gaugeToBuilder = GaugeToBuilder.load(event.params.gauge_);
+  if (gaugeToBuilder == null) return;
+
+  let backerToBuilder = BackerToBuilder.load(
+    event.params.backer_.concat(gaugeToBuilder.builder)
+  );
+  if (backerToBuilder == null) {
+    backerToBuilder = new BackerToBuilder(
+      event.params.backer_.concat(gaugeToBuilder.builder)
+    );
+    backerToBuilder.backer = event.params.backer_;
+    backerToBuilder.builder = gaugeToBuilder.builder;
```

Comment:
If I understood correctly, we should set as Builder the id and in our case the id is the gauge, so it's ok that we set the gauge, but we should either change the name in the `BackerToBuilder` entity or add a comment to specify this is on purpose.

Reply:
No the builder is no longer the gauge is the address, since in the `initializeBuilder` the gauge still doe not exists.